target/ppc/translate/vsx-impl.c.inc

```diff
@@ -2185,6 +2181,49 @@ TRANS(XXBLENDVH, do_xxblendv, MO_16)
 TRANS(XXBLENDVW, do_xxblendv, MO_32)
 TRANS(XXBLENDVD, do_xxblendv, MO_64)
 
+static bool do_xsmaxmin(int t, int a, int b,
+                        void (*helper)(TCGv_ptr, TCGv_ptr, TCGv_ptr, TCGv_ptr))
+{
+    TCGv_ptr ptr_t, ptr_a, ptr_b;
+
+    ptr_t = gen_vsr_ptr(t);
+    ptr_a = gen_vsr_ptr(a);
+    ptr_b = gen_vsr_ptr(b);
+
+    helper(cpu_env, ptr_t, ptr_a, ptr_b);
+
+    tcg_temp_free_ptr(ptr_t);
+    tcg_temp_free_ptr(ptr_a);
+    tcg_temp_free_ptr(ptr_b);
+
+    return true;
+}
```

Comment:
I am thinking if this is the best approach. Should I remove this function and move its body to do_xsmaxmincjdp/do_xsmaxmincqp, changing do_xsmaxmincqp to use gen_avr_ptr?

Reply:
I see two other options:
1 - Receive three `TCGv_ptr` in this method, call `gen_vsr_ptr` in `do_xsmaxmincjdp` and `gen_avr_ptr` in `do_xsmaxmincqp`;
2 - Receive a `TCGv_ptr (*gen_ptr)(int reg)` parameter here.

The first splits resource acquisition and release, but things are quite localized here, it shouldn't be a problem. The second seems overkill for such a simple translation code...